wsd/FileServer.cpp

```diff
@@ -1116,14 +1114,30 @@ void FileServerRequestHandler::preprocessFile(const HTTPRequest& request,
                 << "frame-ancestors " << frameAncestors;
         std::string escapedFrameAncestors;
         Poco::URI::encode(frameAncestors, "'", escapedFrameAncestors);
-        Poco::replaceInPlace(preprocess, std::string("%FRAME_ANCESTORS%"), escapedFrameAncestors);
+        replaceVar["FRAME_ANCESTORS"] = escapedFrameAncestors;
     }
     else
     {
         LOG_TRC("Denied all frame ancestors");
         cspOss << imgSrc << "; ";
     }
 
+    std::string newpreprocess;
+    newpreprocess.reserve(preprocess.length() + 50);
+    std::stringstream checkVar(preprocess);
+    std::string medium;
+
+     while(getline(checkVar, medium, '$'))
+     {
+        if(medium.substr(medium.length()-4, medium.length()) == "<!--")
```

Comment:
`substr` is rather expensive, because it creates a copy. We can improve this (perhaps later).

Also, best to follow the formatting of the file, for consistency. Leave space after the `while` and `if`, and add a line break after `else`. You can also use clang-format on the lines you've changed/added.